handlers/main.yml

```
@@ -20,3 +35,11 @@
     name: icinga2
     state: restarted
   when: i2_manage_service
+
+- name: reload icinga2 on master
+  become: yes
+  service:
+    name: icinga2
+    state: reloaded
+  when: i2_manage_service
+  delegate_to: "{{ groups['masters'][0] }}"
```

I wouldn't want a role to dictate my group names, I'd make this configurable with something like:
```suggestion
  delegate_to: "{{ groups[i2_master_group][0] }}"
```
or
```suggestion
  delegate_to: "{{ i2_master_node }}"
```

---

Thanks, that's a good idea. I was wondering how to solve this issue.
In general there will need to be some clean-up regarding the `i2_master`, `i2_config_master` and group name settings.